pkg/auth/policy.go

```diff
@@ -0,0 +1,268 @@
+package auth
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"golang.org/x/crypto/openpgp"
+
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util"
+)
+
+// A Policy encapsulates the behavior a p2 node needs to authorize
+// its actions. It is possible for implementations to rely on other
+// services for these behaviors, so these calls may be slow or
+// transiently fail.
+type Policy interface {
+	// Check if a pod is authorized to be installed and run on this
+	// node. This involves checking that the pod has a valid signature
+	// and that the signer is authorized to install/run the pod. If
+	// the action is authorized, `nil` will be returned.
+	AuthorizePod(
+		id string,
```

Comment:
does it make more sense to break the cycle? eg move pod.Verify somewhere else, maybe integrate it into the new auth package. if pod needs auth to verify itself, and auth needs pod to determine what to verify, then that indicates (to me) that the division between the packages is poorly defined.